### PR TITLE
nlohmann-json 3.10.0 

### PR DESCRIPTION
**Pull request by amirmasoudabdol**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This updates the `nlohmann-json` formula, and replaces its test with the test provided by @nlohmann. It defaults to build with CMake, and multiple headers.

### CODE REVIEW

File: Formula/nlohmann-json.rb [hunk 1]
```diff
@@ -26,30 +25,73 @@ def install
...
+          }},
           {"list", {1, 0, 2}},
           {"object", {
-            {"happy", true},
-            {"nothing", nullptr}
+            {"currency", "USD"},
+            {"value", 42.99}
           }}
         };
-        std::cout << j << std::endl;
```
Comment (carlocab) [inline Formula/nlohmann-json.rb:45]: Let's keep something like this in the test so we have output to check, please.

**Review by carlocab — COMMENTED** [review]
Thanks, @amirmasoudabdol.

For future reference, please adhere to the [commit style guide](https://docs.brew.sh/Formula-Cookbook) referenced in the PR template when writing commit messages:

> The preferred commit message format for simple version updates is `foobar 7.3` and for fixes is `foobar: fix flibble matrix.`.

File: Formula/nlohmann-json.rb (hunk 1 above)
Comment (amirmasoudabdol) [reply Formula/nlohmann-json.rb]: Sure. I think I will bring back the old test. I realised that it's a better test for the Homebrew.

**Review by carlocab — APPROVED** [review]

**Comment by nlohmann** [comment]
Thanks a lot for taking care of this!

**Pull request closed without merging**